### PR TITLE
X64: Port fpromote to ISLE

### PR DESCRIPTION
**Pull request by jlb6740**

### CODE REVIEW

**Comment by jlb6740** [comment]
Set of modfied files is different from my first attempt of lowering instructions with isle considering lower/isle/generated_code.rs and lower/isle/generated_code.manifest were not modified and are not included in this pr. Looks like a very simple implementation but only ran "cargo test -p wasmtime-cli wast::Cranelift::spec::simd" to verify. Are there other verification steps?

File: cranelift/codegen/src/isa/x64/inst.isle [hunk 1]
```diff
@@ -2812,6 +2812,13 @@
             (_ Unit (emit (MInst.XmmUnaryRmR (SseOpcode.Sqrtpd) x dst))))
         dst))
 
+;; Helper for creating `cvtss2sd` instructions.
+(decl x64_cvtss2sd(Xmm) Xmm)
```
Comment (cfallin) [inline cranelift/codegen/src/isa/x64/inst.isle:2816]: Could you add a space between the term name and the arg list `(Xmm)` here?

File: cranelift/codegen/src/isa/x64/lower.isle [hunk 2]
```diff
@@ -2318,6 +2318,10 @@
 (rule (lower (has_type $F64X2 (sqrt x)))
       (x64_sqrtpd x))
 
+;; Rules for `fpromote` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+(rule (lower (has_type $F64 (sqrt x)))
```
Comment (cfallin) [inline cranelift/codegen/src/isa/x64/lower.isle:2322]: Should this be `fpromote` rather than `sqrt` here?

Comment (jlb6740) [reply cranelift/codegen/src/isa/x64/lower.isle:2322]: Strange .. running the wrong test.

**Pull request merged**